showcases/src/Showcases.ts

```diff
@@ -39,10 +38,12 @@ export class Showcases extends HTMLElement {
       ${acc}
       ${renderComponent({
         componentClass,
-        [showcaseAttr]: showcase,
         hasLongText,
         captionWidth,
         hasCheckeredBackground,
+        ...(!!showcaseClasses
+          ? { showcaseClass: showcase, showcaseStyle: undefined }
+          : { showcaseClass: undefined, showcaseStyle: showcase }),
```

Comment:
looks like TS has a bug here :) `[showcaseAttr]: showcase,` should just work
but to workaround it there is maybe a better alternative:

```ts
      ${renderComponent({
        componentClass,
        showcaseClass: undefined,
        showcaseStyle: undefined,
        [showcaseAttr]: showcase,
        hasLongText,
        captionWidth,
        hasCheckeredBackground,
      })}
```

Reply:
but to fix this properly it's best to call `renderCaptionedBox` and `renderCaptionedText` separately under condition, instead of assigning method to a var